engine/components-api/src/com/cloud/agent/AgentManager.java

```diff
@@ -80,9 +81,9 @@
      *            should the agent stop execution on the first error.
      * @return an array of Answer
      */
-    Answer[] send(Long hostId, Commands cmds) throws AgentUnavailableException, OperationTimedoutException;
+    Answer[] send(Long hostId, Commands cmds) throws AgentUnavailableException, OperationTimedoutException, OperationCancelledException;
 
-    Answer[] send(Long hostId, Commands cmds, int timeout) throws AgentUnavailableException, OperationTimedoutException;
+    Answer[] send(Long hostId, Commands cmds, int timeout) throws AgentUnavailableException, OperationTimedoutException, OperationCancelledException;
```

Comment:
OperationCancelledException is a checked exception. There will be compiler errors if its not thrown by the right methods.